lib/iris/_shapefiles.py

```diff
@@ -3,241 +3,447 @@
+    if not cube.coord(axis="Y", dim_coords=True).has_bounds():
+        cube.coord(axis="Y", dim_coords=True).guess_bounds()
+    # Get the shape of the cube
+    w = len(cube.coord(axis="X", dim_coords=True).points)
+    h = len(cube.coord(axis="Y", dim_coords=True).points)
+    # Get the bounds of the cube
+    # x_bounds = _get_mod_rebased_coord_bounds(cube.coord(x_name))
+    # y_bounds = _get_mod_rebased_coord_bounds(cube.coord(y_name))
+    x_bounds = cube.coord(axis="X", dim_coords=True).bounds
+    y_bounds = cube.coord(axis="Y", dim_coords=True).bounds
+    # Generate Sort-Tile-Recursive (STR) packed R-tree of bounding boxes
+    # https://shapely.readthedocs.io/en/stable/strtree.html
+    grid_boxes = [
+        sgeom.box(x[0], y[0], x[1], y[1]) for y, x in product(y_bounds, x_bounds)
+    ]
+    grid_tree = shapely.STRtree(grid_boxes)
+    # Find grid boxes indexes that intersect with the geometry
+    idxs = grid_tree.query(geometry, predicate="intersects")
+    # Get grid box indexes that intersect with the minimum weight criteria
+    mask_idxs_bool = [
+        grid_boxes[idx].intersection(geometry).area / grid_boxes[idx].area
+        >= minimum_weight
+        for idx in idxs
+    ]
+    mask_idxs = idxs[mask_idxs_bool]
+    mask_xy = [list(product(range(h), range(w)))[i] for i in mask_idxs]
+    # Create mask from grid box indexes
 
-    latitude = latc.name()
-    longitude = lonc.name()
-    return latitude, longitude
+    weighted_mask_template = np.ones((h, w), dtype=bool)
+    # Set mask = True for grid box indexes identified above
+    for xy in mask_xy:
+        weighted_mask_template[xy] = False
+    return weighted_mask_template
 
 
-def _get_mod_rebased_coord_bounds(coord):
-    """Take in a coord and returns a array of the bounds of that coord rebased to the modulus.
+def _make_raster_cube_transform(cube: iris.cube.Cube) -> Affine:
+    """Create a rasterio transform for the cube.
 
-    Parameters
-    ----------
-    coord : :class:`iris.coords.Coord`
-        An Iris coordinate with a modulus.
+    Raises
+    ------
+    CoordinateNotRegularError
+        If the cube dimension coordinates are not regular,
+        such that :func:`iris.util.regular_step` returns an error.
 
     Returns
     -------
-    :class:`np.array`
-        A 1d Numpy array of [start,end] pairs for bounds of the coord.
-
+    :class:`affine.Affine`
+        An affine transform object that maps the geometry domain onto the cube domain.
     """
-    modulus = coord.units.modulus
-    # Force realisation (rather than core_bounds) - more efficient for the
-    #  repeated indexing happening downstream.
-    result = np.array(coord.bounds)
-    if modulus:
-        result[result < 0.0] = (np.abs(result[result < 0.0]) % modulus) * -1
-        result[np.isclose(result, modulus, 1e-10)] = 0.0
-    return result
+    x_points = cube.coord(axis="X", dim_coords=True).points
+    y_points = cube.coord(axis="Y", dim_coords=True).points
```

Comment:
Please assign the coords to variables to avoid running `Cube.coord()` more than you need to (it's slow)